AzureMonitorAgent/ama_tst/modules/install/check_ama.py

```diff
@@ -0,0 +1,86 @@
+import re
+import requests
+import xml.dom.minidom
+
+from error_codes import *
+from errors      import error_info, get_input
+from helpers     import get_package_version
+
+AMA_URL = 'https://docs.microsoft.com/en-us/azure/azure-monitor/agents/azure-monitor-agent-extension-versions'
+
+def get_latest_ama_version(curr_version):
+    try:           
+        r = requests.get(AMA_URL)
+        tbody = r.text.split("<tbody>")[1].split("</tbody>")[0]
+        tbody = "<tbody>" + tbody + "</tbody>"
+        with xml.dom.minidom.parseString(tbody) as dom:
+            rows = dom.getElementsByTagName("tr")
+            for row in rows:
+                cell = row.getElementsByTagName("td")[3]
+                version = cell.firstChild.nodeValue
+                version = re.sub('[A-Za-z ]+', '', version)
+                if (version == ''):
+                    continue
+                if (comp_versions_ge(curr_version, version)):
+                    return (None, None)
+                else:
+                    return (version, None)
+    except Exception as e:
```

Comment:
customers may not necessarily have VM configured to allow connection to `docs.microsoft.com`. If there's an easy way to check this, let's do so. If it is more involved (e.g. we will do connection checks as a separate module in the TST) let's ensure this scenario is described to the user e.g. 
```
Failed to get latest AMA version from Azure documentation at `docs.microsoft.com`: <insert exception here>. Please ensure your network configuration (e.g. firewall, NSG) is configured to allow connectivity to this domain.
```

Reply:
Yes, connection check will be covered in the next module. The case that users cannot connect to docs.microsoft.com will be included.